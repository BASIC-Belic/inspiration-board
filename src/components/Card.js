import React from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';


const Card = (props) => {

  const { text, emoji, id, deleteCardCallback} = props;

  const onDelete = () => {
    deleteCardCallback(id);
  }

  return (
    <div className="card">
      <ul>
        <li> {text} </li>
        <li> {emoji} </li>
        <li><button onClick={onDelete}>Uninspired...</button></li>
      </ul>
    </div>
  )
}

Card.propTypes = {
  text: PropTypes.string.isRequired,
  emoji: PropTypes.string,
  id: PropTypes.number,
  deleteCardCallback: PropTypes.func
};


export default Card;
