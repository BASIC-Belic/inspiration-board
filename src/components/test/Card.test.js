import React from 'react';
import Card from '../Card';
import { shallow } from 'enzyme';

describe('Card', () => {
  test('that it matches an existing snapshot', () => {
    // First Mount the Component in the testing DOM
    // Arrange
    const wrapper = shallow( <
      Card text="text"
      emoji="emoji"
      id={0}
      deleteCardCallback={() => {} }/>);

      // Assert that it looks like the last snapshot
      expect(wrapper).toMatchSnapshot();
    });
  });
