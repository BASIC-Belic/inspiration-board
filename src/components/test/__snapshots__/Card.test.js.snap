// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card that it matches an existing snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    deleteCardCallback={[Function]}
    emoji="emoji"
    id={0}
    text="text"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <li>
                 
                text
                 
        </li>
        <li>
                 
                emoji
                 
        </li>
        <li>
                <button
                        onClick={[Function]}
                >
                        Uninspired...
                </button>
        </li>
</ul>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
             
            text
             
</li>,
          <li>
             
            emoji
             
</li>,
          <li>
            <button
                        onClick={[Function]}
            >
                        Uninspired...
            </button>
</li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "text",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "text",
            " ",
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "emoji",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "emoji",
            " ",
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              onClick={[Function]}
>
              Uninspired...
</button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Uninspired...",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Uninspired...",
            "type": "button",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <li>
                     
                    text
                     
          </li>
          <li>
                     
                    emoji
                     
          </li>
          <li>
                    <button
                              onClick={[Function]}
                    >
                              Uninspired...
                    </button>
          </li>
</ul>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
               
              text
               
</li>,
            <li>
               
              emoji
               
</li>,
            <li>
              <button
                            onClick={[Function]}
              >
                            Uninspired...
              </button>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "text",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "text",
              " ",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                "emoji",
                " ",
              ],
            },
            "ref": null,
            "rendered": Array [
              " ",
              "emoji",
              " ",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                onClick={[Function]}
>
                Uninspired...
</button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Uninspired...",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Uninspired...",
              "type": "button",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
